cmd/print.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/printers"
+)
+
+var (
+	outputFormat = "yaml"
+)
+
+// printCmd represents the print command. It prints ingress-converted
+// HTTPRoutes and Gateways.
+var printCmd = &cobra.Command{
+	Use:   "print",
+	Short: "Prints ingress-converted HTTPRoutes and Gateways",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		resourcePrinter := getResourcePrinter(outputFormat)
+		if resourcePrinter == nil {
+			return fmt.Errorf("%s is not a supported output format", outputFormat)
+		}
+		i2gw.Run(resourcePrinter)
+		return nil
+	},
+}
+
+// // getResourcePrinter returns a specific type of printers.ResourcePrinter
+// based on the provided outputFormat.
+func getResourcePrinter(outputFormat string) printers.ResourcePrinter {
+	switch outputFormat {
+	case "yaml", "":
+		return &printers.YAMLPrinter{}
+	case "json":
+		return &printers.JSONPrinter{}
+	default:
+		return nil
```

Comment:
Any reason not to default to yaml? What would happen if we had a nil resource printer?

Reply:
If I had defaulted to YAML and the users entered "i2gw print -o xml" for example, they would have still received a YAML format without any indication of what was wrong. In this way, if they don't specify anything, the default is YAML (outputFormat=""). Moreover, if they specify an incorrect format, they will receive a comprehensible error message.

Regarding your second point, I wonder if we should return an error when the output is not supported to make this function more intuitive. Or maybe just indicating in the function comment that a nil value may be returned..

Reply:
> I wonder if we should return an error when the output is not supported to make this function more intuitive

Yeah I think that's a good idea, fine with leaving that for a follow up though.